dbms/src/DataStreams/NativeBlockInputStream.cpp

```diff
@@ -57,19 +57,12 @@ NativeBlockInputStream::NativeBlockInputStream(ReadBuffer & istr_, UInt64 server
     }
 }
 
-// also resets few vars from IBlockInputStream (I didn't want to propagate resetParser upthere)
 void NativeBlockInputStream::resetParser()
 {
     istr_concrete = nullptr;
     use_index = false;
-
-#ifndef NDEBUG
-    read_prefix_is_called = false;
-    read_suffix_is_called = false;
-#endif
-
-    is_cancelled.store(false);
-    is_killed.store(false);
+    header.clear();
+    avg_value_size_hints.clear();
```

Comment:
That is important. Otherwise Native read only one msg.

Reply:
Well, it's not clear why this is important. And what this method really does.
As I can see, `istr_concrete` was disabled, but `istr` is still alive and can be used. It's a little bit misleading.
Maybe it's better to add some flag which will be checked later?

Reply:
the red (removed) part is important. Green part is wrong.

It is revert of the that https://github.com/ClickHouse/ClickHouse/pull/8016/commits/3de3ba336a7244f7a26e266b479e7dd86ea5ed52 
 
@abyss7 promised he will incorporate reverted changes back.  


istr_concrete - it may be not needed. header cleaning - also.

Reply:
And that part is now ok: https://github.com/ClickHouse/ClickHouse/pull/8035/commits/f6e059a378393fdb3219b701fa507eac0277a446